self-hosting/README.md

```diff
@@ -0,0 +1,304 @@
+
+Stop containers:
+
+- `docker-compose down or docker stop medic-os && docker stop haproxy`
+
+Remove containers:
+
+- `docker-compose rm or docker rm medic-os && docker rm haproxy`
+
+Clean data volume:
+
+- `docker volume rm medic-data`
+
+Note: Running `docker-compose down -v` from the docker-compose.yml directory would do all the above 3 steps
+
+Prune system:
+
+- `docker system prune`
+
+After following the above commands, you can re-run docker-compose up and create a fresh install (no previous data present)
+
+
+### Port Conflicts
+
+In case you are already running services on HTTP(80) and HTTPS(443), you will have to map new ports to the medic-os container.
+
+Turn off and remove all existing containers that were started:
+
+- `docker-compose down && docker-compose rm`
+To find out which service is using a conflicting port: On Linux:
+
+`sudo netstat -plnt | grep ':<port>'`
+On Mac (10.10 and above):
+
+`sudo lsof -iTCP -sTCP:LISTEN -n -P | grep ':<port>'` 
+
+You can either kill the service which is occupying HTTP/HTTPS ports, or run the container with forwarded ports that are free. In your compose file, change the ports under medic-os:
+
+```yaml
+
+services:
+  medic-os:
+    container_name: medic-os
+    image: medicmobile/medic-os:cht-3.7.0-rc.1
+    volumes:
+      - medic-data:/srv
+    ports:
+     - 8080:80
+     - 444:443
+
+```
+
+Note: You can  substitute 8080, 444 with whichever ports are free on your host. You would now visit https://localhost:444 to visit your project.
+
+### Helpful Docker Commands
+
+#### Get  into a container and view specific service logs 
+
+`docker exec -it medic-os /bin/bash`
+
+##### Once inside container
+
+- view couchdb logs:
+`#less /srv/storage/medic-core/couchdb/logs/startup.log`
+- view medic-api logs:
+`#less /srv/storage/medic-api/logs/medic-api.log`
+- view medic-sentinel logs:
+`#less /srv/storage/medic-sentinel/logs/medic-sentinel.log`
+
+#### View container stderr/stdout log
+
+`# docker logs medic-os`
+`# docker logs haproxy`
+
+#### Other handy commands
+
+- list running containers `#docker ps`
+
+- list all available docker containers with their status `# docker ps -a`
+
+- stop container `# docker stop <container_id>`
+
+- start container `docker start <container_id>`
+
+- list all stoped containers `docker ps -f "status=exited"`
+
+## SSL Certificate Installation
+
+Once your application is up,  you can install your ssl certificates by following the steps below.
+
+- First you need to have your ssl certificate files handy. Both the key file (`ssl.key`) that you sued to generate the certificate and the certificate file (`ssl.crt`) issued by certification authority e.g. letsencrypt. 
+
+### Copy certificates into the medic-os container
+
+```bash
+$sudo docker ps
+$sudo docker cp /path/to/ssl.crt medic-os:/srv/settings/medic-core/nginx/private/ssl.crt
+$docker cp /path/to/ssl.key medic-os:/srv/settings/medic-core/nginx/private/ssl.key
+```
+
+### Edit the nginx configuration file
```

Comment:
I'd change this to "Edit nginx configs and restart" and merge it with the next command `/boot/svc-restart medic-core nginx` to make it more concise.